### PR TITLE
JAMES-3359 Mailbox set rights partial update

### PR DESCRIPTION
**Pull request by LanKhuat**
Related: https://github.com/linagora/james-project/pull/3672

### CODE REVIEW

File: mailbox/api/src/main/java/org/apache/james/mailbox/RightManager.java [hunk 1]
```diff
@@ -135,6 +135,8 @@
      */
     void applyRightsCommand(MailboxPath mailboxPath, MailboxACL.ACLCommand mailboxACLCommand, MailboxSession session) throws MailboxException;
 
+    void applyRightsCommand(MailboxId mailboxId, MailboxACL.ACLCommand mailboxACLCommand, MailboxSession session) throws MailboxException;
```
Comment (chibenwa) [inline mailbox/api/src/main/java/org/apache/james/mailbox/RightManager.java:138]: This method deserves java doc?

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 2]
```diff
@@ -3027,7 +3027,6 @@ trait MailboxSetMethodContract {
          |}""".stripMargin)
   }
 
-  // TODO testing delegation & partial updates
   // TODO write test for rights store manager applyCommandById in MailboxManagerTest
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala]: Can we remove the TODOs from the commit history?

**Review by Arsnael — APPROVED** [review]

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala (hunk 2 above)
Comment (LanKhuat) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala]: outdated?

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala]: No I meant can we remove all traces of their existence?

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala [hunk 3]
```diff
@@ -186,14 +241,39 @@ object IsSubscribedUpdate {
...
+    parseUsername(property)
+      .flatMap(username => parseRights(newValue, property, serializer)
+        .map(rights => SharedWithPartialUpdate(username, rights)))
+
+  def parseUsername(property: String): Either[PatchUpdateValidationException, Username] = try {
+    scala.Right(Username.of(property.substring(MailboxPatchObject.sharedWithPrefix.length)))
+  } catch {
+    case e: Exception => Left(InvalidPropertyException(property, e.getMessage))
+  }
+  def parseRights(newValue: JsValue, property: String, serializer: Serializer): Either[PatchUpdateValidationException, Rfc4314Rights] = serializer.deserializeRfc4314Rights(newValue) match {
```
Comment (chibenwa) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/MailboxSet.scala:255]: We are missing a line break for readability before `def parseRight`

**Review by chibenwa — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Review by Arsnael — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**